src/fulfillments/ed25519.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const nacl = require('tweetnacl')
+const Fulfillment = require('../lib/fulfillment')
+const MissingDataError = require('../errors/missing-data-error')
+
+class Ed25519Fulfillment extends Fulfillment {
+  constructor () {
+    super()
+    this.publicKey = null
+    this.signature = null
+  }
+
+  /**
+   * Write static header fields.
+   *
+   * Some fields are common between the hash and the fulfillment payload. This
+   * method writes those field to anything implementing the Writer interface.
+   * It is used internally when generating the hash of the condition, when
+   * generating the fulfillment payload and when calculating the maximum
+   * fulfillment size.
+   *
+   * @param {Writer|Hasher|Predictor} Target for outputting the header.
+   */
+  writeCommonHeader (writer) {
+    writer.writeVarBytes(this.publicKey)
+  }
+
+  /**
+   * Set the public publicKey.
+   *
+   * This is the Ed25519 public key. It has to be provided as a buffer.
+   *
+   * @param {Buffer} publicKey Public Ed25519 publicKey
+   */
+  setPublicKey (publicKey) {
+    if (!Buffer.isBuffer(publicKey)) {
+      throw new Error('Public key must be a Buffer')
+    } else if (publicKey.length !== 32) {
+      throw new Error('Public key is incorrect length')
+    }
+
+    // TODO Validate public key
```

Comment:
This was mostly a blanket reminder to self. If we don't need any validation, that's great!